packages/scheduled-tasks/src/register.ts

```diff
@@ -0,0 +1,52 @@
+import { container, Plugin, postInitialization, postLogin, preGenericsInitialization, SapphireClient } from '@sapphire/framework';
+import type { ClientOptions } from 'discord.js';
+import { ScheduledTaskHandler } from './lib/ScheduledTaskHandler';
+import { ScheduledTaskStore } from './lib/structures/ScheduledTaskStore';
+import type { ScheduledTasksOptions } from './lib/types';
+
+/**
+ * @since 1.0.0
+ */
+export class ScheduledTasksPlugin extends Plugin {
+	/**
+	 * @since 1.0.0
+	 */
+	public static [preGenericsInitialization](this: SapphireClient, options: ClientOptions): void {
+		container.tasks = new ScheduledTaskHandler(options.tasks);
+	}
+
+	/**
+	 * @since 1.0.0
+	 */
+	public static [postInitialization](this: SapphireClient): void {
+		this.stores.register(new ScheduledTaskStore());
+	}
+
+	/**
+	 * @since 1.0.0
+	 */
+	public static [postLogin](this: SapphireClient): void {
+		container.tasks.createRepeated();
+	}
+}
+
+SapphireClient.plugins.registerPreGenericsInitializationHook(
+	ScheduledTasksPlugin[preGenericsInitialization],
+	'Scheduled-Task-PreGenericsInitialization'
+);
+
+SapphireClient.plugins.registerPostInitializationHook(ScheduledTasksPlugin[postInitialization], 'Scheduled-Task-PostInitialization');
+
+SapphireClient.plugins.registerPostLoginHook(ScheduledTasksPlugin[postLogin], 'Scheduled-Task-PostLogin');
+
+declare module '@sapphire/pieces' {
+	interface Container {
+		tasks: ScheduledTaskHandler;
+	}
+}
+
+declare module 'discord.js' {
+	export interface ClientOptions {
+		tasks?: ScheduledTasksOptions;
+	}
+}
```

Comment:
Move these to `index.ts` then in register add `import './index'`. 

Otherwise the type augments aren't properly loaded when not calling register.